packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderItem/ConditionBuilderItemNumber/ConditionBuilderItemNumber.js

```diff
@@ -2,33 +2,40 @@ import React from 'react';
 
 import { NumberInput } from '@carbon/react';
 
-import { pkg } from '../../../../settings';
 import PropTypes from 'prop-types';
-
-const blockClass = `${pkg.prefix}--condition-builder`;
+import {
+  blockClass,
+  translateWithId,
+} from '../../ConditionBuilderContext/DataConfigs';
 
 export const ConditionBuilderItemNumber = ({
   conditionState,
   config,
   onChange,
 }) => {
   const onChangeHandler = (e, { value }) => {
-    onChange(value + '');
+    if (checkIfValid(value)) {
+      onChange(value + '');
```

Comment:
i believe the extra space is unnecessary since number in `valueRenderers` returns `${val} ${config.unit}` which already has a space

Reply:
That was not extra space, but type casted to string, because we are receiving the number with out unit. But now I did from refactor to append unit from here itself , rather than from valueRenders , there by unit is made available at the root state .